### PR TITLE
[Issue-445] Drop travis build and use github actions for CI pipeline 

### LINKED ISSUE #445: Drop travis build and use github actions for CI pipeline
**Problem description**
 travis-ci.org will be shutting down in several weeks, with all accounts migrating to travis-ci.com
As Pravega has already transitioned to use github actions in https://github.com/pravega/pravega/pull/5563
We need to do this as well

**Problem location**
`.travis.yml`

**Suggestions for an improvement**

### PR DESCRIPTION
**Pull request by thekingofcity**
Signed-off-by: zhongle.wang <zhongle_wang@dell.com>

**Change log description**

* Delete .travis.yml as we will no longer use Travis CI.
* Set up GitHub Actions workflow as follows:
    * Build and run tests on every push and pull requests.
    * Publish the snapshot **ONLY** on the push event of the master branch or the release branch (starts with `r0.x` or `r1.x`).
* Bonus features:
    * Gradle packages are cached, *but for for now we see no speed up ):*.
    * Workflow could be manually triggered via the Github Actions page, as long as the workflow file is in the master or default branch.

**Purpose of the change**

Fixes #445 

**What the code does**

When there is a push or a pull request, trigger the ci/cd to build and run tests.
The process will be:
* Clone the repository.
* Setup the JDK.
* Get the cache, or download the new packages if there is a update in `gradle.properties`.
* Grant execute permission for gradlew.
* Build via Gradle
* Build via Gradle with Scala 2.11
* Publish the snapshot if it is a push on the master or the release branch (starts with r0.x or r1.x).
* Publish the snapshot with Scala 2.11 if it is a push on the master or the release branch (starts with r0.x or r1.x).

**How to verify it**

Please verify that this is functionally equivalent to the previous CI pipeline:

* It compiles the code.
* It executes all unit tests and all integration tests.
* It uploads code coverage correctly.
* For master and release branches, it publishes to JFrog artifactory **ONLY** on push event.
* Triggers are correct and make sense.

### CODE REVIEW

File: .github/workflows/build-artifacts.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+      - name: Build via Gradle with specific Flink and Scala version
+        run: |
+          ./gradlew clean build -PflinkScalaVersion=2.11
+          bash <(curl -s https://codecov.io/bash) -t 9c42ff48-d98f-4444-af05-cf734aa1dbd0
+
+  snapshot:
+    needs: [build]
+    runs-on: ubuntu-latest
+    # only publish the snapshot when it is a push on the master or the release branch (starts with r0.x or r1.x)
+    if: ${{ github.event_name == 'push' && (github.ref == 'refs/heads/master' || startsWith(github.ref, 'refs/heads/r0.') || startsWith(github.ref, 'refs/heads/r1.')) }}
```
Comment (thekingofcity) [inline .github/workflows/build-artifacts.yml:53]: This condition is copied from [Prevega `/.github/workflows/gradle.yml`](https://github.com/pravega/pravega/blob/master/.github/workflows/gradle.yml#L245)

**Comment by thekingofcity** [comment]
We also have to make sure secrets `BINTRAY_KEY` and `BINTRAY_USER` are added in Github Actions secrets [pravega/flink-connectors](https://github.com/pravega/flink-connectors/settings/secrets/actions)

**Comment by thekingofcity** [comment]
The [latest workflow result](https://github.com/thekingofcity/flink-connectors/actions/runs/660262528) of this file.

**Comment by crazyzhou** [comment]
@fpj can you help to review this PR and change the validation checks for this repository?

File: .github/workflows/build-artifacts.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    steps:
+      - name: Publish to repo
+        env:
+          BINTRAY_USER: ${{ secrets.BINTRAY_USER }}
+          BINTRAY_KEY: ${{ secrets.BINTRAY_KEY }}
+        run: ./gradlew publishToRepo -PpublishUrl=jcenterSnapshot -PpublishUsername=$BINTRAY_USER -PpublishPassword=$BINTRAY_KEY
```
Comment (crazyzhou) [inline .github/workflows/build-artifacts.yml:59]: We need to publish both scala 2.11 and 2.12 version connector. So another `./gradlew publishToRepo ... -PflinkScalaVersion=2.11` is needed

File: .github/workflows/build-artifacts.yml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+          # and we will rerun the download to get the newest packages
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+
+      - name: Grant execute permission for gradlew
+        run: chmod +x gradlew
+
+      - name: Build via Gradle
+        run: ./gradlew clean build
+
+      - name: Build via Gradle with specific Flink and Scala version
```
Comment (crazyzhou) [inline .github/workflows/build-artifacts.yml:44]:
```suggestion
      - name: Build via Gradle with Scala 2.11
```

**Review by crazyzhou — CHANGES_REQUESTED** [review]

File: .github/workflows/build-artifacts.yml (hunk 2 above)
Comment (thekingofcity) [reply .github/workflows/build-artifacts.yml:59]: I missed the second version of Scala in snapshot, fixed.

File: .github/workflows/build-artifacts.yml (hunk 3 above)
Comment (thekingofcity) [reply .github/workflows/build-artifacts.yml]: Fixed

**Review by crazyzhou — APPROVED** [review]
LGTM

**Pull request merged**